fluidkeys/secretreceive.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/fluidkeys/fluidkeys/humanize"
+
+	"github.com/fluidkeys/fluidkeys/colour"
+
+	"github.com/fluidkeys/crypto/openpgp"
+	"github.com/fluidkeys/crypto/openpgp/armor"
+	"github.com/fluidkeys/fluidkeys/api"
+	"github.com/fluidkeys/fluidkeys/out"
+	"github.com/fluidkeys/fluidkeys/pgpkey"
+)
+
+func secretReceive() exitCode {
+	out.Print("\n")
+	client := api.NewClient()
+	keys, err := loadPgpKeys()
+
+	if err != nil {
+		printFailed("Couldn't load PGP keys")
+		return 1
+	}
+
+	out.Print(colour.Info("Downloading secrets...") + "\n\n")
+
+	for _, key := range keys {
+		if err != nil {
+			printFailed("Couldn't get email for key " + key.Fingerprint().String() + "\n")
+			continue
+		}
+		encryptedSecrets, err := client.ListSecrets(key.Fingerprint())
+		if err != nil {
+			out.Print("📪 " + displayName(&key) + ": " + colour.Failure(err.Error()) + "\n")
+			continue
+		}
+
+		if len(encryptedSecrets) == 0 {
+			out.Print("📭 " + displayName(&key) + ": No secrets found\n")
+			continue
+		}
+		var passwordPrompter promptForPasswordInterface
```

Comment:
I think you can drop this line. You don't need to reference the interface at all, only sure that the thing you pass *satisfies* that interface, which the `interactivePasswordPrompter{}` does

Reply:
Ah yes!